File: Easy/Array/2535. Difference Between Element Sum and Digit Sum of an Array/Solution.java
class Solution {
    public int differenceOfSum(int[] nums) {
        int elementSum = 0, digitSum = 0;

        for(int num: nums){
            elementSum += num;
            digitSum += digitSum(num);
        }

        return Math.abs(elementSum - digitSum);
    }

    private int digitSum(int n){

        int sum = 0;

        while(n > 0){
            sum += n % 10;
            n /= 10;
        }
        return sum;
    }
}

File: Easy/Array/2535. Difference Between Element Sum and Digit Sum of an Array/Solution.kt
class Solution {
    fun differenceOfSum(nums: IntArray): Int {
        
        val elementSum = nums.sum()

        var digitSum = 0

        for(num in nums) {
            var n = num

            while(n > 0){
                digitSum += n % 10
                n /= 10
            }
        }

        return abs(elementSum - digitSum)
    }
}
